vcap/vcap/testing/input_output_validation.py

```diff
@@ -156,6 +157,12 @@ def _run_inference_on_images(images: List[np.ndarray], capsule: BaseCapsule):
                     ("Capsule failed to output a prediction that matches "
                      "the NodeDescription it had for it's output type. "
                      f"Prediction: {prediction}")
+
+                # Assert the nodes "extra_data" attribute can be JSON encoded
+                # without errors. Typically this can happen if there's a numpy
+                # array carelessly left in the extra_data
+                json.loads(json.dump(output_node.extra_data))
```

Comment:
Should this not be in its own test?

Reply:
It shouldn't. Capsule testing is done through one function, where a host of tests happen on the capsule. 

This function is then exposed through the vcap library under vcap.testing module, which enables repositories to set up easy tests for all of their capsules.

Capsule testing is hence typically done per-capsule. If someone wants to write specific tests for their capsule they can.